x-pack/plugins/cloud_security_posture/public/pages/findings/findings_by_resource_table.test.tsx

```diff
@@ -27,42 +28,140 @@ const getFakeFindingsByResource = () => ({
+    );
+    expect(row).toBeInTheDocument();
+
+    expect(within(row).getByText(item.resource_id)).toBeInTheDocument();
+    expect(within(row).getByText(item.cluster_id)).toBeInTheDocument();
+    expect(within(row).getByText(item.cis_section)).toBeInTheDocument();
+    expect(within(row).getByText(formatNumber(item.failed_findings.total))).toBeInTheDocument();
+    expect(
+      within(row).getByText(new RegExp(numeral(item.failed_findings.normalized).format('0%')))
+    ).toBeInTheDocument();
+  });
+};
+
 describe('<FindingsByResourceTable />', () => {
+  it('renders the error state with the given error message', () => {
+    const error = new Error('some error');
+
+    const props: TableProps = {
+      loading: false,
+      data: undefined,
+      error,
+      ...getPaginationMock(),
+    };
+
+    render(
+      <TestProvider>
+        <FindingsByResourceTable {...props} />
+      </TestProvider>
+    );
+
+    expect(screen.getByText(error.message)).toBeInTheDocument();
+  });
+
   it('renders the zero state when status success and data has a length of zero ', async () => {
     const props: TableProps = {
       loading: false,
-      data: { page: [] },
+      data: { page: [], after: undefined, total: 0 },
       error: null,
+      ...getPaginationMock(),
     };
 
-    render(<FindingsByResourceTable {...props} />);
+    render(
+      <TestProvider>
+        <FindingsByResourceTable {...props} />
+      </TestProvider>
+    );
 
     expect(screen.getByText(TEXT.NO_FINDINGS)).toBeInTheDocument();
   });
 
-  it('renders the table with provided items', () => {
+  it('renders the success state with a table and its provided items', () => {
     const data = Array.from({ length: 10 }, getFakeFindingsByResource);
 
     const props: TableProps = {
       loading: false,
-      data: { page: data },
+      data: { page: data, after: undefined, total: data.length },
       error: null,
+      ...getPaginationMock(),
     };
 
-    render(<FindingsByResourceTable {...props} />);
-
-    data.forEach((item, i) => {
-      const row = screen.getByTestId(
-        TEST_SUBJECTS.getFindingsByResourceTableRowTestId(getResourceId(item))
-      );
-      expect(row).toBeInTheDocument();
-      expect(within(row).getByText(item.resource_id)).toBeInTheDocument();
-      expect(within(row).getByText(item.cluster_id)).toBeInTheDocument();
-      expect(within(row).getByText(item.cis_section)).toBeInTheDocument();
-      expect(within(row).getByText(formatNumber(item.failed_findings.total))).toBeInTheDocument();
-      expect(
-        within(row).getByText(new RegExp(numeral(item.failed_findings.normalized).format('0%')))
-      ).toBeInTheDocument();
-    });
+    render(
+      <TestProvider>
+        <FindingsByResourceTable {...props} />
+      </TestProvider>
+    );
+
+    assertTableDataExists(data);
+  });
+
+  it('calls fetchNext when "next" button is clicked', () => {
```

Comment:
this only tests the right handler is called, not that the `search` function was called with correct pagination params.

i plan on adding this, but a proper tests requires mocking the router and search responses and it got a bit messy, so it was decided to move on to other features  and delay extending the test suite for now.